python/pyspark/rdd.py

```diff
@@ -2029,7 +2030,11 @@ def coalesce(self, numPartitions, shuffle=False):
         >>> sc.parallelize([1, 2, 3, 4, 5], 3).coalesce(1).glom().collect()
         [[1, 2, 3, 4, 5]]
         """
-        jrdd = self._jrdd.coalesce(numPartitions, shuffle)
+        if shuffle:
```

Comment:
Seems you could just call `repartition` here to avoid the code duplication or swap `repartition` to call `coalesce`.